identity_account/src/types/identity_state.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use identity_account_storage::identity::ChainState;
+use identity_iota_core::document::IotaDocument;
+use serde::Deserialize;
+use serde::Serialize;
+
+use crate::error::Error;
+use crate::error::Result;
+
+/// Holds the internal state for the identity.
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+pub struct IdentityState {
+  version: StateVersion,
+  document: Option<Vec<u8>>,
+  chain_state: Option<Vec<u8>>,
+}
+
+impl IdentityState {
+  /// Creates a new [`IdentityState`].
+  pub fn new(document: Option<&IotaDocument>, chain_state: Option<&ChainState>) -> Result<Self> {
+    let document: Option<Vec<u8>> = document
+      .map(|iota_doc| {
+        serde_json::to_vec(iota_doc)
```

Comment:
This can use `ToJson`:

```suggestion
        iota_doc.to_json_vec()
```